### PR TITLE
Allow for manual crafting of Rush! beatmaps with osu! stable

### PR DESCRIPTION
**Pull request by swoolcock**
Adds a separate beatmap converter that allows mappers to craft Rush!-specific beatmaps using the stable osu! editor.  Conversion details are in xmldoc.

I'll include a test map with the next release.

### CODE REVIEW

**Review by frenzibyte — CHANGES_REQUESTED** [review]
I've pushed two tests, one for the beatmap conversion deciding test (whether crafted or generated), and a runnable `BeatmapConversionTest<TConvertValue>`, haven't included an expected result as I don't have much time to look at it, you can check the base class test and the existing test fixtures across legacy rulesets to understand how to make one.

Looks good to go otherwise, really like the fact that I can make a rush! beatmap inside osu!lazer (just need setup screen now), good job on this. 👍

File: osu.Game.Rulesets.Rush/Beatmaps/RushCraftedBeatmapConverter.cs [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    /// 4) Hitsounds are only transferred for minions
+    /// </remarks>
+    public class RushCraftedBeatmapConverter : BeatmapConverter<RushHitObject>
+    {
+        public RushCraftedBeatmapConverter(IBeatmap beatmap, Ruleset ruleset)
+            : base(beatmap, ruleset)
+        {
+        }
+
+        public override bool CanConvert() => true;
```
[the remark this answers is not in the document]
Comment (swoolcock) [reply osu.Game.Rulesets.Rush/Beatmaps/RushCraftedBeatmapConverter.cs:79]: As discussed, I’m not sure if there are performance considerations with using linq there. I figured it was safer to leave it as is.

[the remark this answers is not in the document]
Comment (frenzibyte) [reply osu.Game.Rulesets.Rush/Beatmaps/RushCraftedBeatmapConverter.cs:79]: Oh right forgot about that one, anyways confirmed above that the method is called when a beatmap is started, so should have no performance impact whatsoever.

File: osu.Game.Rulesets.Rush/Beatmaps/RushCraftedBeatmapConverter.cs [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    ///
+    /// Hitobjects may appear anywhere in these sections, but the following positions are suggested
+    /// to make a .osu file more human-readable:
+    /// A: 100,100
+    /// B: 100,200
+    /// C: 100,300
+    /// D: 250,100
+    /// E: 250,300
+    /// F: 400,100
+    /// G: 400,300
```
[the remark this answers is not in the document]
Comment (swoolcock) [reply osu.Game.Rulesets.Rush/Beatmaps/RushCraftedBeatmapConverter.cs:42]: Possibly, but I personally did the entire mapping of the crafted map in VS Code. :P

File: osu.Game.Rulesets.Rush/Beatmaps/RushCraftedBeatmapConverter.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (swoolcock) [reply osu.Game.Rulesets.Rush/Beatmaps/RushCraftedBeatmapConverter.cs:79]: I foresee it being called more than that tbh.  Needs to be able to available at any time for filtering beatmaps.  I'll make this change in the future.

**Pull request merged**